src/app/goals/goals-list/add-goal-button/add-goal-button.component.ts

```diff
@@ -15,6 +15,7 @@ export class AddGoalButtonComponent {
   constructor() {}
 
   showInput() {
+    clearTimeout(this.hidingTimeout);
```

Comment:
I think it doesn't really fix this bug. I think I figured out the cause of bug. Please remove this line and add `*ngIf="isActive"` to input in html. In some cases autofocus directive focus on input after it's already closed, so at the moment we click button again input already focused and blur triggered (because of click on the button) and it immediately closes. The solution is to remove input while state is inactive, which will also fix tab navigation bug.